### PR TITLE
Allow TLS to be configured for calls to remote FHIR servers/Nuts nodes

### PR DESCRIPTION
**Pull request by reinkrul**

### CODE REVIEW

File: domain/transfer/notifier.go [hunk 1]
```diff
@@ -16,10 +17,12 @@ type Notifier interface {
 // FireAndForgetNotifier is a notifier that is optimistic about the receiver's availability.
 // It just sends the notification and assumes the receiver is available.
 type FireAndForgetNotifier struct {
+	TLSConfig *tls.Config
 }
 
 func (f FireAndForgetNotifier) Notify(token, endpoint string) error {
 	response, err := resty.New().
+		SetTLSClientConfig(f.TLSConfig).
```
Comment (woutslakhorst) [inline domain/transfer/notifier.go:25]: what happens if your config is empty?

File: main.go [hunk 2]
```diff
@@ -171,15 +173,29 @@ func registerEHR(server *echo.Echo, config Config, customerRepository customers.
...
+		if tlsClientConfig, err = config.TLS.Client.Load(); err != nil {
+			log.Fatal(err)
+		}
+		fhirNotifier.TLSConfig = tlsClientConfig
+	}
+
+	fhirClientFactory := fhir.NewFactory(
+		fhir.WithURL(config.FHIR.Server.Address),
+		fhir.WithMultiTenancyEnabled(config.FHIR.Server.SupportsMultiTenancy()),
+		fhir.WithTLS(tlsClientConfig),
```
Comment (woutslakhorst) [inline main.go:189]: what happens with a nil pointer here? (eg: no TLS config)

File: domain/transfer/notifier.go (hunk 1 above)
Comment (reinkrul) [reply domain/transfer/notifier.go:25]: sets nil, which is the default value

File: main.go (hunk 2 above)
Comment (reinkrul) [reply main.go:189]: sets nil, which is the default value

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**